demos/multi_channel/object_detection_demo_yolov3/main.cpp

```diff
@@ -108,12 +109,46 @@ static int EntryIndex(int side, int lcoords, int lclasses, int location, int ent
     return n * side * side * (lcoords + lclasses + 1) + entry * side * side + loc;
 }
 
-struct YoloParams {
-    int num;
-    int classes;
-    int coords;
+class YoloParams {
+    template <typename T>
+    void computeAnchors(const std::vector<T> _mask) {
+        std::vector<float> maskedAnchors(num * 2);
+        for (int i = 0; i < num; ++i) {
+            maskedAnchors[i * 2] = anchors[_mask[i] * 2];
+            maskedAnchors[i * 2 + 1] = anchors[_mask[i] * 2 + 1];
+        }
+        anchors = maskedAnchors;
+    }
 
+public:
+    int num = 0, classes = 0, coords = 0;
     std::vector<float> anchors;
+
+    YoloParams() {}
+
+    YoloParams(const std::shared_ptr<ngraph::op::RegionYolo> regionYolo) {
+        coords = regionYolo->get_num_coords();
+        classes = regionYolo->get_num_classes();
+        anchors = regionYolo->get_anchors();
+        auto mask = regionYolo->get_mask();
+        num = mask.size();
+
+        computeAnchors(mask);
+    }
+
+    YoloParams(InferenceEngine::CNNLayer::Ptr layer) {
+        if (layer->type != "RegionYolo")
+            throw std::runtime_error("Invalid output type: " + layer->type + ". RegionYolo expected");
+
+        num = layer->GetParamAsInt("num");
```

Comment:
This is a redundant assignment. I realize that it was there in the original code, but if you're refactoring, you might as well get rid of it.